exchanges/okx/ws_requests.go

```diff
@@ -0,0 +1,364 @@
+	if err != nil {
+		return err
+	}
+
+	single, err := extractSingleItem(resp)
+	if err != nil {
+		return err
+	}
+
+	if !single.Result {
+		return errMassCancelFailed
+	}
+
+	return nil
+}
+
+// WSPlaceSpreadOrder submits a spread order
+func (ok *Okx) WSPlaceSpreadOrder(ctx context.Context, arg *SpreadOrderParam) (*SpreadOrderResponse, error) {
+	if err := arg.Validate(); err != nil {
+		return nil, err
+	}
+
+	id := strconv.FormatInt(ok.Websocket.AuthConn.GenerateMessageID(false), 10)
+
+	var resp []*SpreadOrderResponse
+	if err := ok.SendAuthenticatedWebsocketRequest(ctx, placeSpreadOrderEPL, id, "sprd-order", []SpreadOrderParam{*arg}, &resp); err != nil {
+		return nil, mergeErrorDetails(err, resp)
+	}
+
+	return extractSingleItem(resp)
+}
+
+// WSAmendSpreadOrder amends a spread order
+func (ok *Okx) WSAmendSpreadOrder(ctx context.Context, arg *AmendSpreadOrderParam) (*SpreadOrderResponse, error) {
+	if arg == nil {
+		return nil, fmt.Errorf("%T: %w", arg, common.ErrNilPointer)
+	}
+	if arg.OrderID == "" && arg.ClientOrderID == "" {
+		return nil, order.ErrOrderIDNotSet
+	}
+	if arg.NewPrice == 0 && arg.NewSize == 0 {
+		return nil, errSizeOrPriceIsRequired
+	}
+
+	id := strconv.FormatInt(ok.Websocket.AuthConn.GenerateMessageID(false), 10)
+
+	var resp []*SpreadOrderResponse
+	if err := ok.SendAuthenticatedWebsocketRequest(ctx, amendSpreadOrderEPL, id, "sprd-amend-order", []AmendSpreadOrderParam{*arg}, &resp); err != nil {
+		return nil, mergeErrorDetails(err, resp)
+	}
+
+	return extractSingleItem(resp)
+}
+
+// WsCancelSpreadOrder cancels an incomplete spread order through the websocket connection.
+func (ok *Okx) WsCancelSpreadOrder(ctx context.Context, orderID, clientOrderID string) (*SpreadOrderResponse, error) {
+	if orderID == "" && clientOrderID == "" {
+		return nil, order.ErrOrderIDNotSet
+	}
+
+	arg := make(map[string]string)
+	if orderID != "" {
+		arg["ordId"] = orderID
+	}
+	if clientOrderID != "" {
+		arg["clOrdId"] = clientOrderID
+	}
+
+	id := strconv.FormatInt(ok.Websocket.AuthConn.GenerateMessageID(false), 10)
+
+	var resp []*SpreadOrderResponse
+	if err := ok.SendAuthenticatedWebsocketRequest(ctx, cancelSpreadOrderEPL, id, "sprd-cancel-order", []map[string]string{arg}, &resp); err != nil {
+		return nil, mergeErrorDetails(err, resp)
+	}
+
+	return extractSingleItem(resp)
+}
+
+// WSCancelAllSpreadOrders cancels all spread orders and return success message through the websocket channel.
+func (ok *Okx) WSCancelAllSpreadOrders(ctx context.Context, spreadID string) error {
+	arg := make(map[string]string, 1)
+	if spreadID != "" {
+		arg["sprdId"] = spreadID
+	}
+
+	id := strconv.FormatInt(ok.Websocket.AuthConn.GenerateMessageID(false), 10)
+
+	var resp []*ResponseResult
+	if err := ok.SendAuthenticatedWebsocketRequest(ctx, cancelAllSpreadOrderEPL, id, "sprd-mass-cancel", []map[string]string{arg}, &resp); err != nil {
+		return mergeErrorDetails(err, resp)
```

Comment:
🚧 sprd-mass-cancel can't return errors to merge
If mergeErrorDetalis was just happening to all of them then I think this would be okay

Reply:
Nice spotting!